### PR TITLE
Breaking api change detection

### LINKED ISSUE #220: Public API is not sufficiently tested against breaking changes
## Problem

PR https://github.com/comit-network/comit-js-sdk/pull/183 accidentially introduced a breaking change by removing the InMemoryBitcoinWallet from the public export of the library.
In an effort of releasing fewer breaking changes, 0.15.3 was released which caused the new version to be picked automatically when people started using it through create-comit-app.

## Goal

To not accidentially introduce breaking changes.

## Recommendation

Use some kind of automation to draw attention to the developer if they introduce breaking changes.

Ideas:
- automated tests
- checking TS type declarations into Git and compare them against a new build
- survey other tools in the ecosystem, we can't be the only one solving that problem

### PR DESCRIPTION
**Pull request by da-kami**
* Adds a new command `yarn declarations` to `package.json` that generates the declaration files,
* Excludes generated declarations from linter/prettier/tsc scopes,
* Adds declaration generation to pre-commit hook,
* Checks that any declaration changes are committed in new [GitHub Action](https://github.com/comit-network/comit-js-sdk/runs/665930748?check_suite_focus=true),
* Add an "api-changes" label to the PR if there are any changes in the `declarations/` folder.

Resolves #220.

### CODE REVIEW

**Comment by D4nte** [comment]
Taking over, please note I'll push force.

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -18,3 +18,25 @@ jobs:
           yarn test
         env:
           CI: true
+      - name: Ensure that declarations changes have been committed
+        run: |
+          yarn declarations
+          git status # Exits if there are changes
```
Comment (bonomat) [inline .github/workflows/nodejs.yml:24]: Can you explain or rephrase the comment please? I don't understand it.

Comment (D4nte) [reply .github/workflows/nodejs.yml:24]: `git status` exists if there are changes.

Comment (bonomat) [reply .github/workflows/nodejs.yml:24]: That does not make it better :D What do you mean with _exists_ here please?

I just tried on your branch: 

without changes: 
```
git status
On branch 220-breaking-api-change-detection
Your branch is up to date with 'origin/220-breaking-api-change-detection'.

nothing to commit, working tree clean
```

with changes: 
```
git status
On branch 220-breaking-api-change-detection
Your branch is up to date with 'origin/220-breaking-api-change-detection'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	declarations/ethereum_abi/

nothing added to commit but untracked files present (use "git add" to track)
```

Comment (D4nte) [reply .github/workflows/nodejs.yml]: Check now :)

**Pull request merged**